### PR TITLE
Sign In with actNOW

### PR DESCRIPTION
**Pull request by KobbyMmo**
## Description

Add Act Now Sign In Form

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Desktop Screenshots

https://user-images.githubusercontent.com/23483005/151206650-76da3e8c-5ade-4b85-8f7d-fc1b6ae26482.mp4

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/lib/actnow/index.js [hunk 1]
```diff
@@ -21,6 +21,22 @@ function actnow(site) {
...
+    const response = await fetch(
+      `${process.env.ACTNOW_URL}/v1/accounts/login/`,
+      {
+        method: "POST",
+        headers,
+        body: JSON.stringify(user),
+      }
+    );
+    return response.json();
+  }
```
[the remark this answers is not in the document]
Comment (KobbyMmo) [reply src/lib/actnow/index.js:38]: no doesn't work, and I thought it accepted a rest resources, but it seems it doesn't. 
I might use the password grant flow on the `/token` from [here](https://django-oauth-toolkit.readthedocs.io/en/latest/rest-framework/getting_started.html#step-4-get-your-token-and-use-your-api)  @kilemensi

**Comment by kilemensi** [comment]
Merge conflict @KobbyMmo

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

1. UI is a bit off
S
2. After login, they should be redirected/stay on `/act-now` page (I think `/act-now` page checks if a user is logged in and show a different page)

**Comment by kilemensi** [comment]
> > 2. After login, they should be redirected/stay on `/act-now` page (I think `/act-now` page checks if a user is logged in and show a different page)
> 
> I think this happens already, unfortunately, the redirect URL in the server is pointed to staging hence, it redirects to the staging URL

By server you mean NextJS back-end @KobbyMmo? You should be able to change that if that's the case.

**Comment by KobbyMmo** [comment]
> By server you mean NextJS back-end @KobbyMmo? You should be able to change that if that's the case.

I meant vercel env, but yea, i will change it

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

Somebody needs to get their 👀 checked.

S

PS: In case someone doesn't have access to the designs, [here](https://xd.adobe.com/view/4728124f-a7c3-472c-a8ef-574d5a602886-127b/screen/12b0ed7c-a477-49a8-857b-58a1fddb5ced) sir.

File: src/components/LoginForm/Form.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          />
+          <Button
+            variant="contained"
+            color="secondary"
+            disabled={disabled}
+            onClick={onSubmit}
+            className={classes.authButton}
+          >
+            {fields?.submit?.label}
+          </Button>
```
Comment (kilemensi) [inline src/components/LoginForm/Form.js:71]: Why are we disabling the button? I don't even get the pointer mouse when I hover on the button.

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: src/components/LoginForm/Form.js (hunk 2 above)
Comment (KobbyMmo) [reply src/components/LoginForm/Form.js:71]: if the field isn't valid, I disable the button.
once i get the right data it get enabled
image
Unless you think the ash background should be the state of the button all the time.

Comment (kilemensi) [reply src/components/LoginForm/Form.js:71]: Nah, of Firefox, the page loads with button already disabled @KobbyMmo
Peek 2022-01-26 11-48

**Review by kilemensi — COMMENTED** [review]
How come it signs me in even when I use non-existing email and password? I tried with `Q9tbTzXhjwRS@Q9tbTzXhjwRS.com` and `Q9tbTzXhjwRS` as password.

**Review by kilemensi — COMMENTED** [review]
Well nobody can sign in now (invalid account or existing account)

------------------------

1. Sign-in should obviously redirect/reload the /act-now page,
2. Sign-in error shouldn't redirect/reload any page. It should remain on the same page and show the simple error... something like "Incorrect username and/or password".

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**